File: app/src/main/java/com/the_roberto/kittengifs/EventsTracker.java
package com.the_roberto.kittengifs;

import android.content.Context;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

import static com.google.android.gms.analytics.Logger.LogLevel.ERROR;
import static com.google.android.gms.analytics.Logger.LogLevel.VERBOSE;

public class EventsTracker {
    private static EventsTracker INSTANCE;
    private final GoogleAnalytics googleAnalytics;
    private final Tracker tracker;

    public static void init(Context context) {
        INSTANCE = new EventsTracker(context);
    }

    public static EventsTracker getInstance() {
        return INSTANCE;
    }

    public EventsTracker(Context context) {
        googleAnalytics = GoogleAnalytics.getInstance(context);
        googleAnalytics.getLogger().setLogLevel(BuildConfig.DEBUG ? VERBOSE : ERROR);
        tracker = BuildConfig.DEBUG ? googleAnalytics.newTracker("debug") : googleAnalytics.newTracker(R.xml.global_tracker);
    }

    public void trackNextKitten() {
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory("User Actions")
                .setAction("Next Kitten")
                .build());
    }

    public void trackShare() {
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory("User Actions")
                .setAction("Share")
                .build());
    }

    public void trackSuccessfulKitten() {
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory("Action Results")
                .setAction("Kitten Download")
                .setLabel("success")
                .build());
    }

    public void trackFailedKitten() {
        tracker.send(new HitBuilders.EventBuilder()
                .setCategory("Action Results")
                .setAction("Kitten Download")
                .setLabel("fail")
                .build());
    }
}

File: app/src/main/java/com/the_roberto/kittengifs/EventsTracker.kt
package com.the_roberto.kittengifs

import android.content.Context
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker

object EventsTracker {
    lateinit var googleAnalytics: GoogleAnalytics
    private lateinit var tracker: Tracker

    fun init(context: Context) {
        googleAnalytics = GoogleAnalytics.getInstance(context)
        tracker = if (BuildConfig.DEBUG) googleAnalytics.newTracker("debug") else googleAnalytics.newTracker(R.xml.global_tracker)
    }

    fun trackNextKitten() {
        tracker.send(HitBuilders.EventBuilder().setCategory("User Actions").setAction("Next Kitten").build())
    }

    fun trackShare() {
        tracker.send(HitBuilders.EventBuilder().setCategory("User Actions").setAction("Share").build())
    }

    fun trackSuccessfulKitten() {
        tracker.send(HitBuilders.EventBuilder().setCategory("Action Results").setAction("Kitten Download").setLabel("success").build())
    }

    fun trackFailedKitten() {
        tracker.send(HitBuilders.EventBuilder().setCategory("Action Results").setAction("Kitten Download").setLabel("fail").build())
    }

    fun trackRatingLater(cancel: Boolean) {
        tracker.send(HitBuilders.EventBuilder().setCategory("Rating").setAction("Later").setLabel(if (cancel) "cancel" else "explicit").build())
    }

    fun trackRatingYes() {
        tracker.send(HitBuilders.EventBuilder().setCategory("Rating").setAction("Yes").build())
    }

    fun trackRatingNah() {
        tracker.send(HitBuilders.EventBuilder().setCategory("Rating").setAction("Nah").build())
    }

}
